frontend/amundsen_application/static/js/components/ResourceListItem/MetadataHighlightList/index.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+
+import { TableIcon } from 'components/SVGIcons';
+import { IconSizes } from 'interfaces';
+import { CodeIcon } from 'components/SVGIcons/CodeIcon';
+import { ChartIcon } from 'components/SVGIcons/ChartIcon';
+
+export interface MetadataHighlightListProps {
+  fieldName: string;
+  highlightedMetadataList: string;
+}
+
+const getIcon = (fieldName: string) => {
+  if (fieldName.includes('chart')) {
+    return <ChartIcon size={IconSizes.SMALL} />;
+  }
+  if (fieldName.includes('query')) {
```

Comment:
Let's put 'chart' and 'query' in constants.
We don't usually want any string around our logic.